### PR TITLE
Post view

### PR DESCRIPTION
**Pull request by darya42185**
Implemented PostView and PostPreview
Made sanitize on HTML content of the editor plus content fetched from BE.

## GitHub Board

**Story link**
[#127](https://github.com/ita-social-projects/dokazovi-be/issues/127)

**Issue link**

[#159 PostView + Preview](https://github.com/ita-social-projects/dokazovi-fe/issues/159)

## Summary of change

- [x]  Created PostView
- [x]  Reuse PostView on ArticleCreation preview
- [x]  Sanitize HTML content on the editor and BE fetched post content
- [x]  Add Preview and Publish buttons
- [x]  Implement error 404 page

### CODE REVIEW

File: src/modules/posts/components/PostProfileView.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import { useSelector, useDispatch } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import BorderBottom from '../../../lib/components/Border';
+import LoadingInfo from '../../../lib/components/LoadingInfo';
+import { RootStateType } from '../../../store/rootReducer';
+import { fetchPostById } from '../store/postsSlice';
+import PostInfo from './PostInfo';
+import { useStyles } from '../styles/PostProfileView.styles';
+
+const PostProfileView: React.FC = () => {
```
Comment (SergeOlabin) [inline src/modules/posts/components/PostProfileView.tsx:14]: PostProfileView looks line not a good name.. PostProfice - seems like not applicable one

is this component PostView?

File: src/modules/posts/store/postsSlice.ts [hunk 2]
```diff
@@ -0,0 +1,133 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
```
Comment (SergeOlabin) [inline src/modules/posts/store/postsSlice.ts:5]: fix eslint errors pls

File: src/modules/posts/components/PostProfileView.tsx (hunk 1 above)
Comment (SergeOlabin) [reply src/modules/posts/components/PostProfileView.tsx:14]: seems like this is PostView that we discussed

File: src/store/rootReducer.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
+/* eslint-disable import/named */
```
Comment (SergeOlabin) [inline src/store/rootReducer.ts:1]: why adding this?

File: src/modules/posts/store/postsSlice.ts [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+const initialState: IPostsState = {
+  posts: {
+    posts: [],
+    meta: {
+      loading: LoadingStatusEnum.idle,
+      error: null,
+    },
+  },
+  materials: {} as IMaterialsState,
```
Comment (SergeOlabin) [inline src/modules/posts/store/postsSlice.ts:59]: what are materials used for?

File: src/modules/posts/components/PostProfileView.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  const classes = useStyles();
+  const { postId } = useParams<{ postId: string }>();
+  const {
+    posts,
+    meta: { loading },
+  } = useSelector((state: RootStateType) => state.posts.posts);
+  const dispatch = useDispatch();
```
Comment (SergeOlabin) [inline src/modules/posts/components/PostProfileView.tsx:21]: that's not good..
a single post component should not know anything about all loaded posts.

so let's siplify the stuff here:
- just parse the `id` from the url and get the post and display it
- if there's no post with such an id show 404 error (for now, might be a more informative message latter)
- don't use store to store all loaded posts 
- don't store it in the redux for now

let's not use caching in any form for now, for just displaying the post it looks like obsolete

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: src/modules/postCreation/ArticleCreationPreview.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    (state: RootStateType) => state.newPostDraft.ARTICLE,
+  );
+
+  const articleCreationView = () => {
+    history.push(`/create-article`);
+  };
+
+  return (
+    <Container>
+      <PostPreview draft={draft} />
```
Comment (SergeOlabin) [inline src/modules/postCreation/ArticleCreationPreview.tsx:20]: use PostView here

File: src/modules/posts/components/PostPreview.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (SergeOlabin) [inline src/modules/posts/components/PostPreview.tsx:1]: this is redundant, it is the same as PostView

**Review by SergeOlabin — CHANGES_REQUESTED** [review]

File: debug.log [hunk 8]
```diff
@@ -0,0 +1 @@
+[0115/150555.364:ERROR:directory_reader_win.cc(43)] FindFirstFile: ������� �� ������� ����� ��������� ����. (0x3)
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply debug.log]: fixed

File: src/navigation/router-config.tsx [hunk 9]
```diff
@@ -84,6 +104,13 @@ const ROUTER_CONFIG: IRouterConfig[] = [
     component: NoteCreationView,
     title: 'Створити допис',
   },
+  {
+    path: '/posts/:postId',
+    key: 'POST_PROFILE',
+    exact: true,
+    component: PostView,
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply src/navigation/router-config.tsx:111]: fixed

File: src/modules/posts/components/PostView.tsx [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+
+  return (
+    <Container>
+      <Card>
+        <PostInfo post={post} />
+        <Box>
+          <Typography variant="h4" className={classes.title}>
+            {post.title}
+          </Typography>
+          <Typography className={classes.content}>{post.content}</Typography>
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply src/modules/posts/components/PostView.tsx:22]: fixed

File: src/modules/postCreation/ArticleCreationView.tsx [hunk 11]
```diff
@@ -48,11 +52,20 @@ const ArticleCreationView: React.FC = () => {
...
+        setPostBody({
+          postType: PostTypeEnum.ARTICLE,
+          value: sanitizeHtml(content) as string,
+        }),
+      ); // todo sanitize here
     }, 2000),
     [],
   );
 
+  const articlePreview = () => {
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply src/modules/postCreation/ArticleCreationView.tsx:65]: fixed

File: src/modules/postCreation/ArticleCreationPreview.tsx [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+import PostPreview from '../posts/components/PostPreview';
+import { RootStateType } from '../../store/rootReducer';
+
+const ArticleCreationPreview: React.FC = () => {
+  const history = useHistory();
+  const draft = useSelector(
+    (state: RootStateType) => state.newPostDraft.ARTICLE,
+  );
+
+  const articleCreationView = () => {
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply src/modules/postCreation/ArticleCreationPreview.tsx:14]: fixed

File: src/modules/postCreation/ArticleCreationView.tsx [hunk 13]
```diff
@@ -17,8 +19,10 @@ import {
 } from './store/postCreationSlice';
 import { ICheckboxes, PostTopicSelector } from './PostTopicSelector';
 import { PostTypeEnum } from '../../lib/types';
+import { sanitizeHtml } from '../../lib/utilities/sanitizeHtml';
 
 const ArticleCreationView: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (AntonGluschuk) [reply src/modules/postCreation/ArticleCreationView.tsx:24]: fixed

**Review by SergeOlabin — APPROVED** [review]

**Pull request merged**